FTBFS with a libnormaliz that's using e-antic

e-antic-1.3.0
normaliz-3.9.4
NormalizInterface-1.3.5

* e-antic is a project that mandates at least C++14 due to the use of `std::add_lvalue_reference_t`.
* When normaliz is built with e-antic support, the `<libnormaliz/cone.h>` header will implicitly include `<e-antic/renfxx.h>`, and thereby require that all libnormaliz users have at least C++14 selected, too.
* NormalizInterface has this in configure.ac:
  ```sh
  AX_CXX_COMPILE_STDCXX(11, ,mandatory)
  ```
  which leads to `-std=c++11` being put on the compiler invocation command.

This makes `./configure` fail thus:

```
configure:3667: checking for libnormaliz/cone.h
configure:3667: g++ -std=gnu++11 -c -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g  -I/usr/include conftest.cpp >&5
In file included from /usr/include/e-antic/e-antic.hpp:20,
                 from /usr/include/e-antic/renfxx.h:19,
                 from /usr/include/libnormaliz/general.h:69,
                 from /usr/include/libnormaliz/cone.h:33,
                 from conftest.cpp:48:
/usr/include/e-antic/renf_class.hpp:245:10: error: 'add_lvalue_reference_t' in namespace 'std' does not name a template type; did you mean 'add_lvalue_reference'?
  245 |     std::add_lvalue_reference_t<::renf_t> renf_t() const { return nf; }
      |          ^~~~~~~~~~~~~~~~~~~~~~
[    6s] checking for libnormaliz/cone.h... no
[    6s] configure: error: could not use libnormaliz/cone.h
[    6s] $? = 1
```

--- a/Makefile.in
+++ b/Makefile.in
@@ -4,10 +4,12 @@
 KEXT_NAME = NormalizInterface
 KEXT_SOURCES = src/normaliz.cc
 
-# workaround for building with GAP 4.11 on Cygwin: we need -std=gnu++11
-# instead of -std=c++11 to avoid an error during compilation; see
-# https://github.com/gap-packages/NormalizInterface/pull/91 for details
-KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@  @GMP_CPPFLAGS@ -std=gnu++11
+# Normaliz requires C++11; but some of its dependencies (well at least
+# e-Antic) requires C++14. Since that's been in compilers for a decade
+# now, just always request C++14 support. Note: we use gnu++14 instead
+# of std++14 only for compatibility with GAP versions before 4.12 on
+# Cygwin, see <https://github.com/gap-packages/NormalizInterface/pull/91>.
+KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@  @GMP_CPPFLAGS@ -std=gnu++14
 KEXT_LDFLAGS = @LDFLAGS@ @GMP_LDFLAGS@ -lgmp @NORMALIZ_LDFLAGS@ -lnormaliz -lstdc++
 
 KEXT_USE_AUTOCONF = 1

--- a/configure.ac
+++ b/configure.ac
@@ -19,9 +19,9 @@ AC_PROG_CXX
 AC_LANG([C++])
 
 dnl ##
-dnl ## Check for C++11 support
+dnl ## Check for C++14 support
 dnl ##
-AX_CXX_COMPILE_STDCXX(11, ,mandatory)
+AX_CXX_COMPILE_STDCXX(14, ,mandatory)
 
 dnl ##
 dnl ## Locate the GAP root dir

